src/java.base/share/classes/java/io/ObjectInputStream.java

```diff
@@ -3719,7 +3743,7 @@ private long readUTFSpan(StringBuilder sbuf, long utflen)
             int start = pos;
             int avail = Math.min(end - pos, CHAR_BUF_SIZE);
             // stop short of last char unless all of utf bytes in buffer
-            int stop = pos + ((utflen > avail) ? avail - 2 : (int) utflen);
+            int stop = start + ((utflen > avail) ? avail - 2 : (int) utflen);
```

Comment:
Though reading LV instead of getfield is an improvement, we probably prefer to keep this patch cleaner.